### DISCUSSION TITLE
Curious, but what was your reasoning behind building this library with C++ and WASM?

### DISCUSSION THREAD

**Discussion by anguspiv** [vizzuhq/vizzu-lib#594] [Q&A] [answered]
Hi, 
I just came across your library and its looking beautiful. I was just curious and interested in choosing to build it with C++, and WASM over JavaScript directly. I am assuming it was a performance choice, or better animation quality?

Thank you for building and sharing!

**Comment by simzer** [comment, marked as answer]
Hi! Thanks! Yes, the chart generation is computation-heavy, especially because Vizzu works on bigger data sets than the usual charting libs. This is why we mainly decided to write the core in C++. Also, it makes Vizzu platform independent, it can be integrated into desktop applications as well.
